### PR TITLE
Reorganize about pages and less files

### LINKED ISSUE #8056: Refactor about:about with Aphrodite (part 2)
## Test plan

https://github.com/brave/browser-laptop/pull/10881#issue-256533069

Open about:about

----

Refactor about:about with Aphrodite

Follow up for https://github.com/brave/browser-laptop/issues/7670

### LINKED ISSUE #10700: Remove button.less requirement
**Describe the issue you encountered:** Remove button.less requirement from files where `button` has been replaced with `browserButton`.

- Platform (Win7, 8, 10? macOS? Linux distro?): All

- Brave Version (revision SHA): master branch

- Extra QA steps:
    1. Open `https://expired.badssl.com` (for certerror.js)
    2. Open `https://brave.rules` (for errorPage.js)
    3. Open `about:preferences` (for preferences.js)
    4. Open `http://downloadme.org` (for safebrowsing.js)
    5. Make sure the all error pages are rendered

- Any related issues:

### LINKED ISSUE #10711: Unnecessary Scrollbars on About Pages
## Test plan

#10881

Make sure you do not get the scroll bar on about:brave

----

- Did you search for similar issues before submitting this one?
Yes

- Describe the issue you encountered:
Some _About_ pages contain unnecessary scrollbars.

The problem here is with a `padding-top: 24px` as well as a `height: 100%`. This forces the container to be 24 pixels taller than the window itself. As such, no matter how big your window is, you will always have vertical overflow.

- Platform (Win7, 8, 10? macOS? Linux distro?):
Desktop

- Brave Version (revision SHA):
0.19.1

- Steps to reproduce:
    1. Navigate to `about:brave`
    
- Actual result:
Note presence of unnecessary vertical scroll bars (and often horizontal too).

- Expected result:
Scrollbars are only visible when needed.

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes

- Is this an issue in the currently released version?
Yes

- Can this issue be consistently reproduced?
Yes

- Screenshot if needed:

image

### LINKED ISSUE #10712: Convert brave.js with Aphrodite
## Test plan

https://github.com/brave/browser-laptop/pull/10713#issue-253856185

----

**Describe the issue you encountered:** Convert brave.js with Aphrodite. 

- Platform (Win7, 8, 10? macOS? Linux distro?): All

- Brave Version (revision SHA): master branch

- Extra QA steps:
    1. Open `about:brave`
    2. Make sure the page is rendered

- Any related issues:

### LINKED ISSUE #10880: Reorganize LESS files (import, require, etc) to normalize the design
## Test plan

https://github.com/brave/browser-laptop/pull/10881#issue-256533069

1. Open about:about
2. Open all linked about pages
3. Make sure they are properly styled

----

**Describe the issue you encountered:** I personally think it is important to reorganize less files before refactoring about pages to normalize their design.

Currently some LESS files are imported on other LESS files, others are required on js files. It makes difficult to see the relationship of the LESS files, how they are referred, how the styles are inherited, etc. Without the clear code structure it would be difficult to keep the style consistency (as we don't have automated tests to avoid visual regressions).

- Platform (Win7, 8, 10? macOS? Linux distro?): All

- Brave Version (revision SHA): master branch (for 0.21.x)

- Extra QA steps:
    1.
    2.
    3.

- Any related issues:

### LINKED ISSUE #11120: Normalize section title of about pages
### Description

Normalize section title of about pages.

### PR DESCRIPTION
**Pull request by luixxiul**
**Requires #10951 for buttons on about:autofill.**

Fixes #10711
Closes #8056
Closes #10700
Closes #10712
Closes #10880
Closes #11120

- Add less/about/common.less to the about js files
- Add globalStyles.spacing.aboutPageMargin
- Add the MPL license header
- Remove import from the less files
- Remove button.less from common.less
- Remove require of the less files imported in common.less to avoid duplicates
- Move the height:100% property from common.less to preferences.less

Please have a look at https://github.com/brave/browser-laptop/issues/10880 why I thought this reorganization is important, thanks.

# Test Plans

## 1st commit

Closes #10700
Closes #10880
Addresses #8056

Test Plan 1: `npm run unittest`

Test Plan 2:
1. Open `about:about`
2. Open all linked about pages
3. Make sure they are properly styled
4. Open https://webtorrent.io/free-torrents
5. Open a torrent link
6. Make sure the webtorrent viewer is properly styled

Test Plan 3:
1. Open `https://expired.badssl.com` (for certerror.js)
2. Open `https://brave.rules` (for errorPage.js)
3. Open `about:preferences` (for preferences.js)
4. Open `http://downloadme.org` (for safebrowsing.js)
5. Make sure the all error pages are rendered
6. Make sure all error pages do not have the black background color

Test Plan 4:
1. Open `about:brave`
2. Change the window size
3. Make sure the scroll bar does not appear

## 2nd commit

Fixes #10711
Closes #10712

Test Plan:
1. Open about:brave
2. Make sure the layout is not broken

## 3rd commit

Closes #8056

Test Plan:
1. Open about:about
2. Make sure the page is properly rendered

## 4th commit 

Closes #11120

Test Plan:
1. Open about:about
2. Open about:adblock
3. Open about:autofill
4. Open about:brave
5. Open about:downloads
6. Open about:extensions
7. Make sure the section titles are styled in the same way

----

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/renderer/components/styles/global.js [hunk 1]
```diff
@@ -188,7 +185,13 @@ const globalStyles = {
     overlayButtonMargin: '8px',
     panelMargin: '15px',
     panelItemMargin: '12px',
-    panelPadding: '18px'
+    panelPadding: '18px',
+
+    // about pages
+    aboutPageMargin: '20px',
```
Comment (luixxiul) [inline app/renderer/components/styles/global.js]: Only this line is added. The others are just grouped below.

File: js/about/adblock.js [hunk 2]
```diff
@@ -14,11 +14,11 @@ const SwitchControl = require('../../app/renderer/components/common/switchContro
 const {DefaultTextArea} = require('../../app/renderer/components/common/textbox')
 
 const {StyleSheet, css} = require('aphrodite/no-important')
+const globalStyles = require('../../app/renderer/components/styles/global')
 
 const ipc = window.chrome.ipcRenderer
 
 // Stylesheets
-require('../../less/switchControls.less')
```
Comment (luixxiul) [inline js/about/adblock.js]: This is duping to https://github.com/brave/browser-laptop/pull/10881/files#diff-b79e96c438a5b97dc31f04f68d4817a7R7

File: less/about/history.less [hunk 3]
```diff
@@ -1,19 +1,13 @@
-@import "./siteDetails.less";
```
Comment (luixxiul) [inline less/about/history.less]: deprecated thanks to https://github.com/brave/browser-laptop/pull/10881/files#diff-237402632b025505a0517d846e9e32c1R34

File: less/about/error.less [hunk 4]
```diff
@@ -1,4 +1,8 @@
-@import "./common.less";
```
Comment (luixxiul) [inline less/about/error.less]: deprecated thanks to https://github.com/brave/browser-laptop/pull/10881/files#diff-2369573669cb068e3f220a1c4ef7090cR13 and https://github.com/brave/browser-laptop/pull/10881/files#diff-103b59b981aa5f02b62b3f712ca69c19R12

File: less/about/common.less [hunk 5]
```diff
@@ -13,10 +17,12 @@
   padding: 0;
 }
 
-html, body, #appContainer, #appContainer > div {
-  height: 100%;
-}
```
Comment (luixxiul) [inline less/about/common.less]: deprecated with https://github.com/brave/browser-laptop/pull/10881/files#diff-4b50ff14dadc9756719a577a72f520d6R7

File: app/renderer/about/bookmarks/bookmarks.js [hunk 6]
```diff
@@ -21,6 +21,7 @@ const aboutActions = require('../../../../js/about/aboutActions')
 const windowActions = require('../../../../js/actions/windowActions')
 
 // Stylesheets
+require('../../../../less/about/siteDetails.less')
```
Comment (luixxiul) [inline app/renderer/about/bookmarks/bookmarks.js]: moved from https://github.com/brave/browser-laptop/pull/10881/files#diff-9161cb086bf3eb720aab15d7172f6e4fL1

File: js/about/brave.js [hunk 7]
```diff
@@ -11,16 +11,18 @@ const aboutActions = require('./aboutActions')
...
 const {StyleSheet, css} = require('aphrodite/no-important')
+const globalStyles = require('../../app/renderer/components/styles/global')
 const commonStyles = require('../../app/renderer/components/styles/commonStyles')
 
 const {
   AboutPageSectionTitle,
   AboutPageSectionSubTitle
 } = require('../../app/renderer/components/common/sectionTitle')
 
-require('../../less/about/history.less')
```
Comment (luixxiul) [inline js/about/brave.js]: Removed along with https://github.com/brave/browser-laptop/pull/10881/files#diff-6f6a63435cccded794332a0616b5f0e3L7

File: js/about/certerror.js [hunk 8]
```diff
@@ -9,8 +9,8 @@ const messages = require('../constants/messages')
 const ipc = window.chrome.ipcRenderer
 const {isSourceAboutUrl, getTargetAboutUrl} = require('../lib/appUrlUtil')
 
-require('../../less/button.less')
```
Comment (luixxiul) [inline js/about/certerror.js]: Duping to https://github.com/brave/browser-laptop/pull/10881/files#diff-b79e96c438a5b97dc31f04f68d4817a7R6

File: js/about/errorPage.js [hunk 9]
```diff
@@ -8,8 +8,8 @@ const BrowserButton = require('../../app/renderer/components/common/browserButto
 const appActions = require('../../js/actions/appActions')
 const tabActions = require('../../app/common/actions/tabActions')
 
-require('../../less/button.less')
```
Comment (luixxiul) [inline js/about/errorPage.js]: Deprecated with https://github.com/brave/browser-laptop/pull/10881/files#diff-b79e96c438a5b97dc31f04f68d4817a7R6

File: js/about/history.js [hunk 10]
```diff
@@ -30,6 +30,8 @@ const {
 } = require('../../app/renderer/components/common/sectionTitle')
 
 // Stylesheets
+require('../../less/about/common.less')
+require('../../less/about/siteDetails.less')
```
Comment (luixxiul) [inline js/about/history.js]: Added as this is no longer imported on history.less

File: js/about/preferences.js [hunk 11]
```diff
@@ -63,10 +63,9 @@ const ipc = window.chrome.ipcRenderer
 const hintCount = 3
 
 // Stylesheets
-require('../../less/switchControls.less')
```
Comment (luixxiul) [inline js/about/preferences.js]: Removed based on https://github.com/brave/browser-laptop/pull/10881/files#diff-b79e96c438a5b97dc31f04f68d4817a7R7 on common.less

File: js/about/preferences.js [hunk 12]
```diff
@@ -63,10 +63,9 @@ const ipc = window.chrome.ipcRenderer
...
-require('../../less/about/preferences.less')
 require('../../less/forms.less')
-require('../../less/button.less')
```
Comment (luixxiul) [inline js/about/preferences.js]: Ditto on https://github.com/brave/browser-laptop/pull/10881/files#diff-b79e96c438a5b97dc31f04f68d4817a7R6

File: js/about/safebrowsing.js [hunk 13]
```diff
@@ -7,8 +7,8 @@ const BrowserButton = require('../../app/renderer/components/common/browserButto
 
 const {StyleSheet, css} = require('aphrodite/no-important')
 
-require('../../less/button.less')
```
Comment (luixxiul) [inline js/about/safebrowsing.js]: Removed based on https://github.com/brave/browser-laptop/pull/10881/files#diff-b79e96c438a5b97dc31f04f68d4817a7R6

File: less/about/autofill.less [hunk 14]
```diff
@@ -1,4 +1,8 @@
-@import "./common.less";
```
Comment (luixxiul) [inline less/about/autofill.less]: Removed as it is required on autofill.js See: https://github.com/brave/browser-laptop/pull/10881/files#diff-f3189199f630a8432fae752aa453b2ccR15

File: less/about/bookmarks.less [hunk 15]
```diff
@@ -1,4 +1,8 @@
-@import "./siteDetails.less";
```
Comment (luixxiul) [inline less/about/bookmarks.less]: Moved to https://github.com/brave/browser-laptop/pull/10881/files#diff-c74077737ab01e5191e480529919c489R24

File: less/about/newtab.less [hunk 16]
```diff
@@ -2,22 +2,15 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+@import "../variables.less";
 @import "../animations.less";
-@import "./common.less";
```
Comment (luixxiul) [inline less/about/newtab.less]: Moved to https://github.com/brave/browser-laptop/pull/10881/files#diff-491dd03a03308cec37fad2ad0dc7ba67R13 on newprivatetab.js and https://github.com/brave/browser-laptop/pull/10881/files#diff-dbb992f7cd2508b46d5cd98e609be5a5R36 on newtab.js

File: less/about/preferences.less [hunk 17]
```diff
@@ -1,12 +1,11 @@
-@import "./common.less";
```
Comment (luixxiul) [inline less/about/preferences.less]: Moved to https://github.com/brave/browser-laptop/pull/10881/files#diff-e3eeb751016b2ce9f8278efce585a461R67 on preferences.js

File: less/about/siteDetails.less [hunk 18]
```diff
@@ -1,16 +1,10 @@
...
 @import "./itemList.less";
 @import "./common.less";
 
-strong, div, span, li, em, p, a {
-  font-weight: 400;
-  -webkit-font-smoothing: antialiased;
-}
-
-body {
-  background: @veryLightGray;
```
Comment (luixxiul) [inline less/about/siteDetails.less]: Deprecated with https://github.com/brave/browser-laptop/pull/10881/files#diff-b79e96c438a5b97dc31f04f68d4817a7R22 on common.less. As common.less is required on all of the about js files, it should not cause a regression.

File: less/webtorrent.less [hunk 19]
```diff
@@ -2,9 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-@import "./variables.less";
-@import "./button.less";
-@import "./about/siteDetails.less";
+@import "variables.less";
```
Comment (luixxiul) [inline less/webtorrent.less]: Deprecated with https://github.com/brave/browser-laptop/pull/10881/files#diff-9774044dc3cb63ab522b79039669a42aR17 and https://github.com/brave/browser-laptop/pull/10881/files#diff-9774044dc3cb63ab522b79039669a42aR18

**Comment by luixxiul** [comment]
Adding the impact/high label as there is a possibility of visual regressions.

File: js/about/about.js [hunk 20]
```diff
@@ -6,48 +6,43 @@ const React = require('react')
...
 const {StyleSheet, css} = require('aphrodite/no-important')
 const globalStyles = require('../../app/renderer/components/styles/global')
-const commonStyles = require('../../app/renderer/components/styles/commonStyles')
 
 const {
   AboutPageSectionTitle,
   AboutPageSectionSubTitle
 } = require('../../app/renderer/components/common/sectionTitle')
 
-require('../../less/about/history.less')
```
Comment (luixxiul) [inline js/about/about.js]: Removed as `siteDetails*` classes are no longer applied.

**Comment by cezaraugusto** [comment]
why blocked?

**Comment by luixxiul** [comment]
@cezaraugusto because this requires #10901 to make sure that the section sub title would be displayed as a block: https://github.com/brave/browser-laptop/pull/10881/commits/fc9a3e0c7680156ab214bb547f264cf529258317#diff-943709aef6ee258bb12a2cbb33018988R24 (to close #8056)

**Comment by cezaraugusto** [comment]
ok #10901 merged pls rebase

File: less/about/siteDetails.less [hunk 21]
```diff
@@ -1,14 +1,8 @@
-@import "./itemList.less";
```
Comment (luixxiul) [inline less/about/siteDetails.less]: Deprecated with:
  - https://github.com/brave/browser-laptop/pull/10881/files#diff-c74077737ab01e5191e480529919c489R28 on bookmarks.js
  - https://github.com/brave/browser-laptop/pull/10881/files#diff-83a9c31b7f8db399cce791535a519b5fR25 on brave.js
  - https://github.com/brave/browser-laptop/pull/10881/files#diff-237402632b025505a0517d846e9e32c1R34 on history.js
  - https://github.com/brave/browser-laptop/pull/10881/files#diff-9774044dc3cb63ab522b79039669a42aR18 on webtorrent/entry.js

**Comment by luixxiul** [comment]
Rebased. Also I added https://github.com/brave/browser-laptop/pull/10881/commits/cb3392623630457921aacf5380de8e1bc7faffac to add `fitContent` to browserButton as an option.

**Comment by luixxiul** [comment]
Requires #10951 for buttons on about:autofill

- https://github.com/brave/browser-laptop/pull/10951/files#diff-f3189199f630a8432fae752aa453b2ccL221
- https://github.com/brave/browser-laptop/pull/10951/files#diff-f3189199f630a8432fae752aa453b2ccL231

**Comment by luixxiul** [comment]
@cezaraugusto this PR is blocked by https://github.com/brave/browser-laptop/pull/10951 (for buttons on about:autofill). I would like to see #10951 reviewed before this one 😄

**Comment by cezaraugusto** [comment]
left a comment in #10951 the text inside buttons doesn't seem ok to me, mind checking there what's happening?

**Comment by luixxiul** [comment]
Let me close the PR until I'll find a better fix on https://github.com/brave/browser-laptop/pull/10951.

**Pull request closed without merging**